### ISSUE TITLE
EntityRef::entity panics when using an empty Entity received from World::entity

### ISSUE THREAD

**Issue by dbartussek** [Ralith/hecs#184]
This code panics with the message `'index out of bounds: the len is 64 but the index is 4294967295'`:

```
use hecs::*;

fn main() {
    let mut world = World::new();
    world.spawn(());

    let entities: Vec<Entity> = world.iter().map(|e| e.entity()).collect();

    for e in entities {
        let entity_ref = world.entity(e).unwrap();
        println!("{:?}", e);
        entity_ref.entity();
    }
}
```

Digging into it, the issue is caused by the [check in Entities::get line 405](https://github.com/Ralith/hecs/blob/5eaf664ad3beb3b83f4c4dd6a41d05327368aa6c/src/entities.rs#L405) because apparently the id of the empty archetype is 0.

As a result, empty entites are kind of Schrödinger's entities. They exist and can be iterated, but attempting to access them by their id will fail because they are seen as 'pending'.

There is a [comment in ArchetypeSet::new](https://github.com/Ralith/hecs/blob/5eaf664ad3beb3b83f4c4dd6a41d05327368aa6c/src/world.rs#L1094) that says archetype 0 always represents empty entities, so I am not sure what the intended behaviour should be and another comment on [World::flush](https://github.com/Ralith/hecs/blob/5eaf664ad3beb3b83f4c4dd6a41d05327368aa6c/src/world.rs#L759) states that reserved entities are converted into empty entities, so there seems to be some reason for this behaviour, but if intentional it is not documented that empty entities are not supported and leads to a rather confusing crash.

World::flush and the check in Entities::get were created in commit https://github.com/Ralith/hecs/commit/90c0197d6c8feb286b3fe1aa8381fe9fe11a0745.

**Comment by Ralith** [comment]
Thanks for the report! On first examination, I believe this due to some missed changes in 220e53f53576299baf54aa1d7c8209dde68a6fb4. #185 contains a prospective fix, which resolves the panic. However, I need to take some time to verify this.

**Comment by Ralith** [comment]
Fix published in 0.6.4.
